### PR TITLE
Fix window where sshd is enabled with less secure configuration - #49

### LINKED ISSUE #49: appears to have window where sshd is enabled with less secure configuration
Reading ansible-sshd, the "sensible" defaults disable password authentication, for example.  But this configuration is not imposed until _after_ sshd is installed.

See [http://unix.stackexchange.com/questions/321621/configuring-my-sshd-securely-with-automation](http://unix.stackexchange.com/questions/321621/configuring-my-sshd-securely-with-automation). If the system crashed or lost power between the two steps, it could start later with the insecure configuration, and could run like that for some time.

dpkg-divert can be used, for an example using ansible see https://github.com/debops-contrib/ansible-etckeeper/blob/master/tasks/main.yml

### PR DESCRIPTION
**Pull request by sourcejedi**
Fix #49

### CODE REVIEW

File: files/scripts/dpkg-state [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	proc.stdout.close()
+
+	# Process should either have returned 0,
+	# or been killed because we closed the pipe (stdout).
+	# Otherwise, we had an error.
+	proc.wait()
+	if proc.returncode not in [0, -signal.SIGPIPE]:
+		return None
+
+	return state
```
Comment (muelli) [inline files/scripts/dpkg-state]: minor nitpick: state may not necessarily be initialised if proc.stdout does not contain the line of desire.  How about to raise in the else clause of the for loop?

File: tasks/main.yml [hunk 2]
```diff
@@ -3,6 +3,60 @@
 - name: Pre hooks
   include: '{{ lookup("task_src", "sshd/pre_main.yml") }}'
 
+- name: Check if OpenSSH is installed (or half-installed)
+  script: scripts/dpkg-state openssh-server
+  register: sshd__prev_pkg_state
```
Comment (drybjed) [inline tasks/main.yml]: This variable should be named `sshd__register_prev_pkg_state` to keep the correct variable namespace.

**Comment by sourcejedi** [comment]
Thanks, here's the revised version.

**Comment by drybjed** [comment]
@sourcejedi Do you plan to modify the script as well?

**Comment by sourcejedi** [comment]
I modified the script to use `else:`, as per my comment.

**Pull request merged**